src/tactic/nth_rewrite/default.lean

```diff
@@ -0,0 +1,122 @@
+do rewrites ← q.rules.mmap $ λ r, unpack_rule r >>= nth_rewrite e,
+   rewrites.join.nth n <|> fail "failed: not enough rewrites found"
+
+/-- Rewrite the `n`th occurence of the rewrite rules `q` of (optionally after zooming into) a
+hypothesis or target `h` which is an application of a relation. -/
+meta def get_nth_rewrite_with_zoom
+  (n : ℕ) (q : rw_rules_t) (path : list expr_lens.dir) (h : option expr) : tactic tracked_rewrite :=
+do e ← target_or_hyp_type h,
+   (ln, new_e) ← expr_lens.entire.zoom path e,
+   rw ← get_nth_rewrite n q new_e,
+   return ⟨ln.fill rw.exp, rw.proof >>= ln.congr, rw.addr.map $ λ l, path ++ l⟩
+
+/-- Rewrite the `n`th occurence of the rewrite rules `q` (optionally on a side)
+at all the locations `loc`. -/
+meta def nth_rewrite_core (path : list expr_lens.dir) (n : parse small_nat) (q : parse rw_rules)
+  (l : parse location) : tactic unit :=
+do let fn := λ h, get_nth_rewrite_with_zoom n q path h
+                    >>= λ rw, (rw.proof >>= replace_in_state h rw.exp),
+   match l with
+   | loc.wildcard := l.try_apply (fn ∘ some) (fn none)
+   | _            := l.apply     (fn ∘ some) (fn none)
+   end,
+   tactic.try (tactic.reflexivity reducible),
+   (returnopt q.end_pos >>= save_info <|> skip)
+
+namespace interactive
+
+open expr_lens
+
+/-- `nth_write n rules` performs only the `n`th possible rewrite using the `rules`.
+
+The core `rewrite` has a `occs` configuration setting intended to achieve a similar
+purpose, but this doesn't really work. (If a rule matches twice, but with different
+values of arguments, the second match will not be identified.)
+
+See also: `nth_rewrite_lhs` and `nth_rewrite_rhs` -/
+meta def nth_rewrite
+  (n : parse small_nat) (q : parse rw_rules) (l : parse location) : tactic unit :=
+nth_rewrite_core [] n q l
+
+/-- `nth_write_lhs n rules` performs only the `n`th possible rewrite using the `rules`,
+but only working on the left hand side.
+
+The core `rewrite` has a `occs` configuration setting intended to achieve a similar
+purpose, but this doesn't really work. (If a rule matches twice, but with different
+values of arguments, the second match will not be identified.)
+
+See also: `nth_rewrite` and `nth_rewrite_rhs` -/
+meta def nth_rewrite_lhs (n : parse small_nat) (q : parse rw_rules) (l : parse location) : tactic unit :=
+nth_rewrite_core [dir.F, dir.A] n q l
+
+/-- `nth_write_rhs n rules` performs only the `n`th possible rewrite using the `rules`,
+but only working on the right hand side.
+
+The core `rewrite` has a `occs` configuration setting intended to achieve a similar
+purpose, but this doesn't really work. (If a rule matches twice, but with different
+values of arguments, the second match will not be identified.)
+
+See also: `nth_rewrite` and `nth_rewrite_lhs` -/
+meta def nth_rewrite_rhs (n : parse small_nat) (q : parse rw_rules) (l : parse location) : tactic unit :=
+nth_rewrite_core [dir.A] n q l
+
```

Comment:
```suggestion

/-- ... -/
add_tactic_doc
{ name       := "erewrite / erw",
  category   := doc_category.tactic,
  decl_names := [``nth_rewrite, ``nth_rewrite_lhs, ``nth_rewrite_rhs],
  tags       := ["rewriting"] }
```

You can write a new doc string that captures all three variants. Alternatively, update the `nth_rewrite` doc string to (briefly) explain the variants, drop the doc string above `add_tactic_doc`, and add `inherit_description_from := ``nth_rewrite`. I prefer the latter option!

Reply:
Done.

Reply:
It's not perfect that we now have the same doc-string written out three times. I agree that it gives a good result for the end-user in terms of pop-up help, but it's harder to maintain/read the source.

Reply:
That said, I don't have a better solution.

Reply:
Oh, I do. I added a user command `copy_doc_string`, and used that.

Reply:
Perhaps someone else should chime in as to whether this is a reasonable solution.